lib/ansible/modules/network/fortimanager/fmgr_fwobj_service.py

```diff
@@ -0,0 +1,702 @@
+        }
+        # SET DELETE URL
+        url = '/pm/config/adom/{adom}/obj/firewall/service/category' \
+              '/{name}'.format(adom=paramgram["adom"], name=paramgram["category"])
+
+    if paramgram["mode"] == "set":
+        response = fmg.set(url, datagram)
+        return response
+        # IF MODE = ADD  -- USE THE 'ADD' API CALL MODE
+    if paramgram["mode"] == "add":
+        response = fmg.add(url, datagram)
+        return response
+        # IF MODE = DELETE  -- USE THE DELETE URL AND API CALL MODE
+    if paramgram["mode"] == "delete":
+        response = fmg.delete(url, datagram)
+        return response
+
+
+def main():
+    argument_spec = dict(
+        adom=dict(required=False, type="str"),
+        host=dict(required=True, type="str"),
+        password=dict(fallback=(env_fallback, ["ANSIBLE_NET_PASSWORD"]), no_log=True),
+        username=dict(fallback=(env_fallback, ["ANSIBLE_NET_USERNAME"]), no_log=True),
+        mode=dict(required=False, type="str", choices=['add', 'set', 'delete'], default="add"),
+
+        app_category=dict(required=False, type="str"),
+        app_service_type=dict(required=False, type="str"),
+        application=dict(required=False, type="str"),
+        category=dict(required=False, type="str"),
+        check_reset_range=dict(required=False, type="str"),
+        color=dict(required=False, type="int"),
+        comment=dict(required=False, type="str"),
+        custom_type=dict(required=False, type="str", choices=['tcp_udp_sctp', 'icmp', 'icmp6', 'ip', 'http', 'ftp',
+                                                              'connect', 'socks_tcp', 'socks_udp', 'all'],
+                         default="all"),
+        explicit_proxy=dict(required=False, type="str", choices=['enable', 'disable'], default="disable"),
+        fqdn=dict(required=False, type="str"),
+        group_name=dict(required=False, type="str"),
+        group_member=dict(required=False, type="str"),
+        icmp_code=dict(required=False, type="int"),
+        icmp_type=dict(required=False, type="int"),
+        iprange=dict(required=False, type="str"),
+        name=dict(required=False, type="str"),
+        protocol=dict(required=False, type="str"),
+        protocol_number=dict(required=False, type="int"),
+        sctp_portrange=dict(required=False, type="str"),
+        session_ttl=dict(required=False, type="int"),
+        object_type=dict(required=False, type="str", choices=['custom', 'group', 'category']),
+        tcp_halfclose_timer=dict(required=False, type="int"),
+        tcp_halfopen_timer=dict(required=False, type="int"),
+        tcp_portrange=dict(required=False, type="str"),
+        tcp_timewait_timer=dict(required=False, type="int"),
+        udp_idle_timer=dict(required=False, type="int"),
+        udp_portrange=dict(required=False, type="str"),
+        visibility=dict(required=False, type="str"),
+
+    )
+
+    module = AnsibleModule(argument_spec, supports_check_mode=True, )
```

Comment:
This module claims support for `check_mode` but never seems to have any conditionals against `module.check_mode`.